aws/resource_aws_lambda_permission_test.go

```diff
@@ -570,6 +594,42 @@ EOF
 }
 `
 
+var testAccAWSLambdaPermissionConfig_withStatementIdPrefix = `
+resource "aws_lambda_permission" "with_statement_id_prefix" {
+    statement_id_prefix = "AllowExecutionWithStatementIdPrefix-"
+    action = "lambda:InvokeFunction"
+    function_name = "${aws_lambda_function.test_lambda.arn}"
+    principal = "events.amazonaws.com"
+}
+
+resource "aws_lambda_function" "test_lambda" {
+    filename = "test-fixtures/lambdatest.zip"
+    function_name = "lambda_function_name_perm"
+    role = "${aws_iam_role.iam_for_lambda.arn}"
+    handler = "exports.handler"
+    runtime = "nodejs4.3"
+}
+
+resource "aws_iam_role" "iam_for_lambda" {
+    name = "%s"
```

Comment:
I'm assuming that this will be failing as this is not being replaced

Reply:
Indeed:

```
--- FAIL: TestAccAWSLambdaPermission_withStatementIdPrefix (1.28s)
    testing.go:503: Step 0 error: Configuration is invalid.
        
        Warnings: []string(nil)
        
        Errors: []string{"aws_iam_role.iam_for_lambda: \"name\" must match [\\w+=,.@-]"}
```